pgen-native/parsetok.h

```diff
@@ -0,0 +1,66 @@
+
+/* Parser-tokenizer link interface */
+
+#ifndef Py_PARSETOK_H
+#define Py_PARSETOK_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "node.h"
+
+typedef struct {
```

Comment:
move this to its own file? If you're targeting C++, we should use exceptions anyway...